syntax = "proto3"; 

package adaptive_resource_manager; 

service AdaptiveResourceManager {
    rpc ResourceExchange(ResourceExchangeRequest) returns (ResourceExchangeResponse); 
}

message MicroserviceData {
    string microservice_name = 1; 
    string scaling_action = 2; 
    int32 desired_replicas = 3; 
    int32 current_replicas = 4; 
    int32 cpu_request = 5; 
    int32 max_replicas = 6; 
    double cpu_percentage = 7; 
}

message ResourceExchangeDecision {
    string microservice_name = 1; 
    string scaling_action = 2; 
    int32 desired_replicas = 3; 
    int32 max_replicas = 4; 
    int32 cpu_request = 5; 

}

message ResourceExchangeRequest {
    repeated MicroserviceData data = 1; 
}


message ResourceExchangeResponse {
    repeated ResourceExchangeDecision decision = 1; 
    
}
